src/test/java/com/meilisearch/integration/SearchTest.java

```diff
@@ -315,5 +315,58 @@ public void testSearchMatches() throws Exception {
 		assertEquals(375, res_gson.hits[0].getMatchesInfo().get("overview").get(2).start);
 		assertEquals(3, res_gson.hits[0].getMatchesInfo().get("overview").get(2).length);
 	}
-	
+	/**
+	 * Test place holder search
+	 */
+	@Test
+	public void testPlaceHolder() throws Exception {
+		String indexUid = "placeHolder";
+		Index index = client.index(indexUid);
+		GsonJsonHandler jsonGson = new GsonJsonHandler();
+
+		TestData<Movie> testData = this.getTestData(MOVIES_INDEX, Movie.class);
+		UpdateStatus updateInfo = jsonGson.decode(
+			index.addDocuments(testData.getRaw()),
+			UpdateStatus.class
+		);
+
+		index.waitForPendingUpdate(updateInfo.getUpdateId());
+
+		Results res_gson = jsonGson.decode(
+			index.search(""),
+			Results.class
+		);
+		assertEquals(index.search(""), index.search(new SearchRequest(null)));
+		assertEquals(20, res_gson.hits.length);
+		assertEquals("419704", res_gson.hits[0].getId());
+		assertEquals("Ad Astra", res_gson.hits[0].getTitle());
+		assertEquals("920", res_gson.hits[10].getId());
+		assertEquals("Cars", res_gson.hits[10].getTitle());
+		assertEquals("420817", res_gson.hits[19].getId());
+		assertEquals("Aladdin", res_gson.hits[19].getTitle());
```

Comment:
Checking these exact hardcoded values may seem to make sense at first sight, but it can introduce a few problems that may pop up in the future. As MeiliSearch Core continues to evolve, changes in the way the MeiliSearch algorithm works may end up breaking these tests, even if the SDK functionality is still OK (imagine for example that de default order of the ranking rules changes, and so, the order of the results is not the same. But the Placeholder search is still working, right? well according to this test, no. This is the complexity that we are introducing here). So from my POV, we could make this more "generic". We need to know that call works, that it returns results, that we have the default amount of them (20)... but checking the specific ID is always risky. I know we do it sometimes in some other tests, but in this case doesn't seem to be really necessary, WDYT?

Reply:
Tbh I was a little confused on how I had to prove that placeholder was working, but now that you mention it, it makes a lot of sense testing it without the hardcoded values.
Checking only the amount of results seems like a good approach. Thank you for the feedback!